tests/UnifiedSpecTests/UnifiedTestRunner.php

```diff
@@ -213,93 +215,41 @@ private function doTestCase(stdClass $test, string $schemaVersion, array $runOnR
+                $this->getTopology(),
+                $this->getServerParameters(),
+                $this->isAuthenticated(),
+                $this->isServerless(),
+            ];
+        }
 
         foreach ($runOnRequirements as $o) {
             $runOnRequirement = new RunOnRequirement($o);
-            if ($runOnRequirement->isSatisfied($serverVersion, $topology, $serverParameters)) {
+            if ($runOnRequirement->isSatisfied(...$cachedIsSatisfiedArgs)) {
                 return;
             }
         }
 
         // @todo Add server parameter requirements?
-        Assert::markTestSkipped(sprintf('Server version "%s" and topology "%s" do not meet test requirements', $serverVersion, $topology));
-    }
-
-    /**
-     * Return the server parameters (cached for subsequent calls).
-     */
-    private function getCachedServerParameters() : stdClass
-    {
-        static $cachedServerParameters;
-
-        if (isset($cachedServerParameters)) {
-            return $cachedServerParameters;
-        }
-
-        $cachedServerParameters = $this->getServerParameters();
-
-        return $cachedServerParameters;
-    }
-
-    /**
-     * Return the server version (cached for subsequent calls).
-     */
-    private function getCachedServerVersion() : string
-    {
-        static $cachedServerVersion;
-
-        if (isset($cachedServerVersion)) {
-            return $cachedServerVersion;
-        }
-
-        $cachedServerVersion = $this->getServerVersion();
-
-        return $cachedServerVersion;
-    }
-
-    /**
-     * Return the topology type (cached for subsequent calls).
-     *
-     * @throws UnexpectedValueException if topology is neither single nor RS nor sharded
-     */
-    private function getCachedTopology() : string
-    {
-        static $cachedTopology = null;
-
-        if (isset($cachedTopology)) {
-            return $cachedTopology;
-        }
-
-        switch ($this->getPrimaryServer()->getType()) {
-            case Server::TYPE_STANDALONE:
-                $cachedTopology = RunOnRequirement::TOPOLOGY_SINGLE;
-                break;
-
-            case Server::TYPE_RS_PRIMARY:
-                $cachedTopology = RunOnRequirement::TOPOLOGY_REPLICASET;
-                break;
-
-            case Server::TYPE_MONGOS:
-                $cachedTopology = $this->isShardedClusterUsingReplicasets()
-                    ? RunOnRequirement::TOPOLOGY_SHARDED_REPLICASET
-                    : RunOnRequirement::TOPOLOGY_SHARDED;
-                break;
-
-            default:
-                throw new UnexpectedValueException('Toplogy is neither single nor RS nor sharded');
-        }
-
-        return $cachedTopology;
+        Assert::markTestSkipped(sprintf(
+            'Server (version=%s, toplogy=%s, auth=%s) does not meet test requirements',
+            $cachedIsSatisfiedArgs[0],
+            $cachedIsSatisfiedArgs[1],
+            $cachedIsSatisfiedArgs[3] ? 'yes' : 'no'
+        ));
```

Comment:
I didn't bother to add serverless here, which I think is fine because it's not really supported yet. We can consider doing so when the functionality is actually implemented.

That said, I don't see a nice way to print the server parameters in this message -- since there could be tons of them. One approach might be trying to log the environment details when we first compute/cache them, and then keeping the skip message short and avoiding redundant output. I'm not familiar with what, if any, logging mechanisms PHPUnit makes available.